Lindhart.Analyser.MissingAwaitWarning/Lindhart.Analyser.MissingAwaitWarning/Lindhart.Analyser.MissingAwaitWarning.Test/TestData.cs

```diff
@@ -1,40 +1,52 @@
 ﻿namespace Lindhart.Analyser.MissingAwaitWarning.Test
 {
-	internal class TestData
-	{
-		/// <summary>
-		/// File for testing diagnosis
-		/// </summary>
-		public const string TestDiagnosis = @"
+    internal class TestData
+    {
+        /// <summary>
+        /// File for testing diagnosis
+        /// </summary>
+        public const string TestDiagnosis = @"
+using System;
 using System.Threading.Tasks;
+
 namespace AsyncAwaitGames
 {
-    // In my real case, that method just returns Task.
     public interface ICallee { Task<int> DoSomethingAsync(); }
 
-    public class Callee: ICallee
+    public class Callee : ICallee
     {
-        public async Task<int> DoSomethingAsync() => await Task.FromResult(0); // Should not give a warning
+        public async Task<int> DoSomethingAsync() => await Task.FromResult(0);
     }
+
     public class Caller
     {
         public void DoCall()
         {
             ICallee xxx = new Callee();
 
-            // In my real case, the method just returns Task,
-            // so there is no type mismatch when assigning a result 
-            // either.
-            xxx.DoSomethingAsync(); // Should give a warning.
-
-            var task = xxx.DoSomethingAsync(); // Should give a warning when strict rule enabled
-            xxx.DoSomethingAsync().Result; // Should not give a warning
-
-            xxx.DoSomethingAsync().ConfigureAwait(false); // Should give a warning
+            // Should not give a warning
+            var _ = xxx.DoSomethingAsync().Result;
+            xxx.DoSomethingAsync().Wait();
```

Comment:
After you made this build I had a long look at this test, and found perhaps the following should not give a warning (and as such should be added to the lines that "Should not give a warning"):
`Func<Task> test = () => xxx.DoSomethingAsync();`

I'd like to hear your thoughts about it.

Reply:
I think this case lies along with my question in the general conversation:
"I also think now that the new code for anonymous functions should go under the strict option as there're scenarios when this can be considered too aggressive. What do you say?"

This is one of the scenarios where warnings may be unwanted as if you await test() later it will be fine. But you may also want to await tasks in all pass-though functions as well. That's why I suggest to move all lambda-related cases under the strict option.

Reply:
Good point!

Reply:
I've moved now everything related to lambdas under the strict option. Some of the cases should actually always give warnings without a doubt but I don't know how to reliably distinguish all the cases. So, I think it's better to take in this solution for now and then if possible improve it later.

Reply:
Good idea.

I've just checked the code and 

I think, that in the future I'll divide the strict rule into two separate rules with better descriptions.